tonic/models/vic/grid_params.py

```diff
@@ -728,8 +939,20 @@ def make_grid(grid_file, soil_file, snow_file, vegl_file, veg_file,
                      spatial_snow, july_tavg_supplied,
                      veglib_fcan, veglib_photo, blowing_snow,
                      vegparam_lai, vegparam_fcan, vegparam_albedo,
-                     lai_src, fcan_src, alb_src)
-        return nc_file
+                     lai_src, fcan_src, alb_src, carbon)
+        if state_file and nc_state_file:
+            write_nc_state(nc_state_file, target_attrs, target_grid,
+                           grid_dict['soil_dict'], grid_dict['snow_dict'],
+                           grid_dict['veg_dict'], grid_dict['lake_dict'],
+                           grid_dict['state_dict'], version_in,
+                           organic_fract, spatial_frost,
+                           spatial_snow, july_tavg_supplied,
+                           veglib_fcan, veglib_photo, blowing_snow,
+                           vegparam_lai, vegparam_fcan, vegparam_albedo,
+                           lai_src, fcan_src, alb_src, carbon)
+            return [nc_file, nc_state_file]
+        else:
+            return [nc_file, None]
```

Comment:
No need for the brackets in these lines.  The function will return a tuple based on the presence of the comma.